skimage/filters/thresholding.py

```diff
@@ -899,10 +881,10 @@ def _mean_std(image, w):
         kern[indices] = (-1) ** (image.ndim % 2 != np.sum(indices) % 2)
 
     total_window_size = np.prod(w)
-    sum_full = ndi.correlate(integral, kern, mode='constant')
-    m = crop(sum_full, pad_width) / total_window_size
-    sum_sq_full = ndi.correlate(integral_sq, kern, mode='constant')
-    g2 = crop(sum_sq_full, pad_width) / total_window_size
+    sum_full = correlate_sparse(integral, kern, mode='valid')
+    m = sum_full / total_window_size
+    sum_sq_full = correlate_sparse(integral_sq, kern, mode='valid')
+    g2 = sum_sq_full / total_window_size
```

Comment:
@rfezzani over here 👆. If we don't have mode valid, we pad a potentially very large volume with a potentially very large pad, and compute a bunch of useless values, only to remove them at the end.

Reply:
I see, now my next question is: is the acceleration coming from the kernel sparseness or from the `"valid"` mode? Can't we then simply use `scipy.signal.correlate` here instead of `ndi.correlate`?

Reply:
No, it's the sparseness. For large kernels, SciPy (rightly) switches to FFT, which is much more expensive than 4 (2D) or 8 (3D) simple passes over the data.